quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -289,12 +293,20 @@ impl S3CompatibleObjectStorage {
             .byte_stream()
             .await
             .map_err(|io_error| Retry::Permanent(StorageError::from(io_error)))?;
-        self.s3_client
+        let mut put_object_request = self.s3_client
             .put_object()
             .bucket(bucket)
             .key(key)
             .body(body)
-            .content_length(len as i64)
+            .content_length(len as i64);
+        if let Some(_encryption) = &self.server_side_encryption {
+            put_object_request = match _encryption.as_str() {
+                "Aes256" => put_object_request.server_side_encryption(ServerSideEncryption::Aes256),
+                "AwsKms" => put_object_request.server_side_encryption(ServerSideEncryption::AwsKms),
```

Comment:
@rdettai 
Thanks for your review, I've edited server_side_encryption as an enum variable, and added it in the multipart upload, I've also added the kms key id variable.

I tested it by setting the bucket policy to deny all objects that aren't encrypted, if i didn't set server_side_encryption it will access denied.  I also tested encryption with AES-256, AWS KMS, and AWS KMS DSS. When the objects were uploaded to S3, I checked the object properties and found them to be as expected.

Regarding the KMS key ID, it is used with AWS KMS and AWS KMS DSS. If a specific KMS ID is provided, that key will be used. If not, AWS-managed keys will be used.

Reply:
@rdettai I've added SSE to the multipart upload, but I'm unsure what file size would be appropriate for testing. I tested with a 5GB file—would that be sufficient?

Reply:
It's more about checking that at least one of the files created by Quickwit is larger than the multipart limit, and that that file received the proper encryption settings.

Reply:
Hi @rdettai 
I have updated the code to the multipart upload and tested both single-part and multipart uploads. The results are as expected. 
The data is encrypted according to my configurations, and uploads are rejected if they aren't encrypted as per my bucket's policy. 
Here is some of my bucket policy that denies all objects without an encryption.

```
        {
            "Sid": "VisualEditor0",
            "Effect": "Deny",
            "Principal": {
                "AWS": "arn:aws:iam::123456789:root"
            },
            "Action": "s3:PutObject",
            "Resource": [
                "arn:aws:s3:::my-bucket/*"
            ],
            "Condition": {
                "StringNotEquals": {
                    "s3:x-amz-server-side-encryption": [
                        "AES256",
                        "aws:kms",
                        "aws:kms:dsse"
                    ]
                }
            }
        },
        {
            "Sid": "VisualEditor1",
            "Effect": "Deny",
            "Principal": {
                "AWS": "arn:aws:iam::123456789:root"
            },
            "Action": "s3:PutObject",
            "Resource": [
                "arn:aws:s3:::my-bucket/*"
            ],
            "Condition": {
                "Null": {
                    "s3:x-amz-server-side-encryption": "true"
                }
            }
        }